staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -558,7 +558,7 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		// the user matches the version of latest storage object.
 		resourceVersion, err := e.Storage.Versioner().ObjectResourceVersion(obj)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, kubeerr.NewBadRequest(fmt.Sprintf("Failed to parse resource version of %s: %v", name, err))
```

Comment:
just include the error: `fmt.Sprintf("failed to parse resourceVersion: %v", err)`

Reply:
Done